### PR TITLE
Fix custom element creation

### PR DESCRIPTION
**Pull request by thornbill**
**Changes**
Fix the creation of webcomponents using `setAttribute()`

**Issues**
See the comments in https://github.com/jellyfin/jellyfin-web/pull/3601

### CODE REVIEW

File: src/controllers/dashboard/plugins/repositories/index.js [hunk 1]
```diff
@@ -64,8 +67,7 @@ function getRepositoryElement(repository) {
     const listItem = document.createElement('div');
     listItem.className = 'listItem listItem-border';
 
-    const repoLink = document.createElement('a');
-    repoLink.setAttribute('is', 'emby-linkbutton');
+    const repoLink = createCustomElement('a', 'emby-linkbutton');
     repoLink.className = 'clearLink listItemIconContainer';
```
Comment (dmitrylyzo) [inline src/controllers/dashboard/plugins/repositories/index.js:71]:
```suggestion
    repoLink.classList.add('clearLink', 'listItemIconContainer');
```

File: src/controllers/dashboard/plugins/repositories/index.js [hunk 2]
```diff
@@ -90,9 +92,8 @@ function getRepositoryElement(repository) {
 
     listItem.appendChild(body);
 
-    const button = document.createElement('button');
+    const button = createCustomElement('button', 'paper-icon-button-light');
     button.type = 'button';
-    button.setAttribute('is', 'paper-icon-button-light');
     button.className = 'btnDelete';
```
Comment (dmitrylyzo) [inline src/controllers/dashboard/plugins/repositories/index.js:97]:
```suggestion
    button.classList.add('btnDelete');
```

File: src/scripts/libraryMenu.js [hunk 3]
```diff
@@ -667,8 +671,7 @@ import Headroom from 'headroom.js';
         if (customMenuOptions) {
             getMenuLinks().then(links => {
                 links.forEach(link => {
-                    const option = document.createElement('a');
-                    option.setAttribute('is', 'emby-linkbutton');
+                    const option = createCustomElement('a', 'emby-linkbutton');
                     option.className = 'navMenuOption lnkMediaFolder';
```
Comment (dmitrylyzo) [inline src/scripts/libraryMenu.js:675]:
```suggestion
                    option.classList.add('navMenuOption', 'lnkMediaFolder');
```

File: src/elements/helpers.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const createCustomElement = (tag, typeExtension) => {
+    if (window.WebComponents) {
+        const elem = document.createElement(tag);
+        elem.setAttribute('is', typeExtension);
+        return elem;
+    }
```
Comment (dmitrylyzo) [inline src/elements/helpers.js:6]: In Chrome 70 `window.WebComponents` is defined (`{"flags":{"log":{}}}`), but it works without this block - need to find a valid test or maybe this block is redundant.

Comment (thornbill) [reply src/elements/helpers.js:6]: The `setAttribute` method was required for it to work for me in modern Firefox.

Comment (dmitrylyzo) [reply src/elements/helpers.js:6]: `document.createElement` method works fine in:
- Chrome 79
- Chrome 100
- Firefox 72
- Firefox 99
- webOS 1.2 emulator
- webOS 4 emulator

Dunno what about Safari

File: src/controllers/dashboard/plugins/repositories/index.js (hunk 1 above)
Comment (thornbill) [reply src/controllers/dashboard/plugins/repositories/index.js:71]: Good call. This is exactly what was tripping me up in my testing! :tada:

**Review by crobibero — APPROVED** [review]

**Review by dmitrylyzo — APPROVED** [review]

**Pull request merged**